### ISSUE TITLE
Option for assets to export at document canvas size

### ISSUE THREAD

**Issue by prolost** [adobe-photoshop/generator-assets#164]
I'd like the ability to force generated assets to maintain the PSD canvas bounds.

In VFX for film, we use Photoshop to create texture maps for 3D models and layered matte paintings for digital environments. Generator is the perfect tool for automating these multi-layered exports.

But we absolutely need the assets to retain the PSD canvas dimensions so that they'll remain in registration when imported into the 3D or compositing program. There's a very large community of users who could take advantage of this. A similar request is the third comment on [this blog post](http://blogs.adobe.com/samartha/2013/09/a-closer-look-at-the-photoshop-generator-syntax.html#comment-152750).

What I propose is a simple tag to force the uncropped export. It would automatically trim off any pixels outside the canvas area and pad the assets to the same bounds as the canvas.

It would work with the exiting scale syntax. Here are some examples:

**canvas foo.jpg**

**canvas 50% foo.png32**

**canvas 1920x? foo.png**

Not sure what the best tag would be. Maybe _canvas,_ or _uncropped,_ or _docsize._ Or maybe this can be merged with other related requests for more padding and image-bounds export options.

**Comment by JimCollinson** [comment]
Perhaps adding a layer mask to a 'default' tagged layer could set the bounds of the document, for all layers below to be cropped to on generation?

**Comment by Jakobud** [comment]
Is there any solution for this? Having individual layers be generated to assets without respecting the document bounds makes it COMPLETELY USELSS for any layers that extend beyond the bounds.

**Comment by JimCollinson** [comment]
@Jakobud Have you tried putting a mask on each layer to be generated? That's what I now do, and it works.

Alternatively there are these options within the generator.js config file:

```
"clip-all-images-to-document-bounds":false,
"clip-all-images-to-artboard-bounds":false,
```

Which you could experiment with; I've not had cause to use them yet.

**Comment by Jakobud** [comment]
Yeah I have been using the mask route. But honestly it is just a work around to a problem that shouldn't exist. I mean, if your document bounds define the intended size of your picture, why in the world would you EVER, in any circumstances whatsoever want to export any layer assets bigger than your document bounds. 

Hopefully this is a bug that gets fixed someday.

**Comment by JimCollinson** [comment]
Here are the details:
https://github.com/adobe-photoshop/generator-core/wiki/Generator-Configuration-File-Format

It's worth giving it a go. I'm in and out of that fill all day FWIW. It seems bizarre that we have to do that, these options should really be in a toolbar or in the prefs. Feels hacky to me, and not great for a product used by people with wide ranging technical abilities.

I can, however, envisage circumstances where you would want to generate assets that live outside the bounds in some complex designs and webpage builds, so I would expect it to be a configurable option, just probably not the default.
